generators/generator-microsoft-bot-adaptive/generators/app/templates/dotnet/webapp/Controllers/BotController.cs

```diff
@@ -1,34 +1,63 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Bot.Builder;
 using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.Integration.Runtime.Settings;
 
 namespace <%= botName %>.Controllers
 {
     // This ASP Controller is created to handle a request. Dependency Injection will provide the Adapter and IBot
     // implementation at runtime. Multiple different IBot implementations running at different endpoints can be
     // achieved by specifying a more specific type for the bot constructor argument.
-    [Route("api/messages")]
     [ApiController]
     public class BotController : ControllerBase
     {
-        private readonly IBotFrameworkHttpAdapter _adapter;
-
+        private readonly Dictionary<string, IBotFrameworkHttpAdapter> _adapters = new Dictionary<string, IBotFrameworkHttpAdapter>();
         private readonly IBot _bot;
 
-        public BotController(IBotFrameworkHttpAdapter adapter, IBot bot)
+        public BotController(
+            IEnumerable<IBotFrameworkHttpAdapter> adapters, 
+            IEnumerable<AdapterSettings> adapterSettings, 
+            IBot bot)
         {
-            this._adapter = adapter;
-            this._bot = bot;
+            _bot = bot ?? throw new ArgumentNullException(nameof(bot));
+
+            foreach (var adapter in adapters ?? throw new ArgumentNullException(nameof(adapters)))
+            {
+                var settings = adapterSettings.FirstOrDefault(s => s.Enabled && s.Name == adapter.GetType().FullName);
+
+                if (settings != null)
+                {
+                    _adapters.Add(settings.Route, adapter);
+                }
+            }
         }
 
         [HttpPost]
         [HttpGet]
-        public async Task PostAsync()
+        [Route("api/{route}")]
```

Comment:
Detail for the future: there are reserved words that route cannot take. Most of them enforced by asp.net core routing. Examples: `action`, `area`, `controller`, `handler`, `page`. On our side, `skills` is a reserved word as well. There is a point to be made about adding validations for these in composer where folks configure adapters. However that would happen in the runtime plugin so that composer doesn't have asp.net specific code. Furthermore, we could add another layer of validation here, forbidding registering adapters for those routes. Thoughts? Not a pri 0 or 1 but nice to have for R13.

Reply:
Agreed on this validation not being tied to ASP.NET specific implementation. We probably want to discuss validations all up - I think there's a broad category of improvements we should make around that area, and this would fit into that bucket in my mind.